API/plumber.R

```diff
@@ -18,34 +19,44 @@ stats <- pins::pin_get("alex.gold/bike_station_info", board = "rsconnect")
 #* @param station_id the id number of (a) station(s) in the Capitol Bikeshare program
 #* @param max_time time to stop predictions
 #* @param interval prediction interval
-#* @param which which model, defaults to rxgb
 #* @get /pred
-function(station_id, max_time = 86400, interval = 600, which = "r_xgb") {
+function(station_id, max_time = 86400, interval = 600) {
   # sanitize inputs
   station_id <- as.numeric(station_id)
-  if (!all(station_id %in% stats$station_id)) stop("That station does not exist.")
+  if (!all(station_id %in% stations$station_id)) stop("That station does not exist.")
 
+  # select model and recipe from pin
+  recipe <- model_details$recipe
+  model <- model_details$mod
+
+  # create interval for prediction
   max_time <- as.numeric(max_time)
   interval <- as.numeric(interval)
 
   times <- Sys.time() + seq(0, max_time, by = interval)
 
-  df <- tidyr::crossing(times, station_id = as.character(station_id)) %>%
```

Comment:
I see you don't like namespacing fns. Any particular reason? I've always preferred so that someone reading can know where it comes from.

Reply:
I find it harder to read ... guess its a matter of preference 😬